examples/msw/mocks/index.js

```diff
@@ -0,0 +1,15 @@
+const { rest } = require("msw");
+const { setupServer } = require("msw/node");
+
+const handlers = [
+  rest.get("https://my-mock-api.com", (_, res, ctx) => {
+    return res(ctx.status(200), ctx.json({ message: "from msw" }));
+  }),
+];
+
+const server = setupServer(...handlers);
+server.listen({ onUnhandledRequest: "warn" });
+console.info("MSW initialised");
+
+process.once("SIGINT", () => server.close());
+process.once("SIGTERM", () => server.close());
```

Comment:
To be fair, if the process is terminated there's no way the `server` would keep doing anything. Keep in mind that despite the name, there are no actual HTTP servers establishes. `setupServer` is a plain function that listens to intercepted events and handles them.

Perhaps you had some other scenarios in mind for these process events?

Reply:
@kettanaito @Girish21 It seems that headers aren't being read properly during unhandled requests. 


```sh
[MSW] Expected response resolver to return a mocked response Object, but got undefined. The original response is going to be used instead.

...

Response {
  size: 0,
  timeout: 0,
  [Symbol(Body internals)]: {
    body: PassThrough {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 5,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: true,
      [Symbol(kCapture)]: false,
      [Symbol(kCallback)]: null
    },
    disturbed: false,
    error: null
  },
  [Symbol(Response internals)]: {
    url: 'https://randomuser.me/api',
    status: 200,
    statusText: 'OK',
    headers: Headers { [Symbol(map)]: [Object: null prototype] },
    counter: 0
  }
}
```

in interceptors: `ClientRequest/utils/getIncomingMessageBody.ts` and `ClientRequest/NodeClientRequest.ts`

Reply:
@kentcdodds, what do you think about adding the listeners?

Reply:
I've never done it and I don't know what situation would necessitate this. I think it's unnecessary

Reply:
This setup makes sense if `server` spawned a child process, or any other side-effect detached from the current process' thread. That is not the case, so if the process exists/terminates it will also destroy the execution context for `server`. 

You can safely remove this.